backend/src/sql/list-dreps.sql

```diff
@@ -29,22 +29,88 @@ DRepStatus AS (
   FROM
     drep_hash dh
     LEFT JOIN (
-      SELECT
+      -- Replace ROW_NUMBER with DISTINCT ON
+      SELECT DISTINCT ON (dr.drep_hash_id)
         dr.id,
         dr.drep_hash_id,
-        dr.deposit,
-        ROW_NUMBER() OVER (PARTITION BY dr.drep_hash_id ORDER BY dr.tx_id DESC) AS rn
+        dr.deposit
       FROM
         drep_registration dr
       WHERE
         dr.deposit IS NOT NULL
+      ORDER BY dr.drep_hash_id, dr.tx_id DESC
     ) AS dr_deposit ON dr_deposit.drep_hash_id = dh.id
-    AND dr_deposit.rn = 1
     LEFT JOIN tx ON tx.id = dr_deposit.id
     LEFT JOIN block ON block.id = tx.block_id
     LEFT JOIN block AS block_first_register ON block_first_register.id = tx.block_id
     CROSS JOIN DRepActivity
   GROUP BY dh.id, dr_deposit.deposit, DRepActivity.epoch_no, DRepActivity.drep_activity
+),
+LatestDRepRegistrations AS (
+  SELECT DISTINCT ON (dr.drep_hash_id)
+    dr.id,
+    dr.drep_hash_id,
+    dr.deposit,
+    dr.voting_anchor_id,
+    dr.tx_id,
+    tx.hash AS tx_hash
+  FROM
+    drep_registration dr
+    JOIN tx ON tx.id = dr.tx_id
+  ORDER BY dr.drep_hash_id, dr.tx_id DESC
+),
+NonDeregisterVotingAnchor AS (
+  -- Replace ROW_NUMBER with DISTINCT ON
+  SELECT DISTINCT ON (dr.drep_hash_id)
+    dr.id,
+    dr.drep_hash_id,
+    dr.voting_anchor_id,
+    tx.hash AS tx_hash
+  FROM
+    drep_registration dr
+    JOIN tx ON tx.id = dr.tx_id
+  WHERE 
+    dr.deposit IS NOT NULL
+    AND dr.deposit >= 0
```

Comment:
```suggestion
    AND COALESCE(dr.deposit, 0) >= 0
```
As empty deposit is the update transaction